src/snowflake/snowpark/session.py

```diff
@@ -577,6 +580,12 @@ def __init__(
             self._conn._get_client_side_session_parameter(
                 _PYTHON_SNOWPARK_AUTO_CLEAN_UP_TEMP_TABLE_ENABLED, False
             )
+            and pkg_resources.parse_version(self.version)
+            >= pkg_resources.parse_version(
+                self._conn._get_client_side_session_parameter(
+                    _PYTHON_SNOWPARK_AUTO_CLEAN_UP_TEMP_TABLE_ENABLED_VERSION, "0.0.0"
```

Comment:
let's make the default empty, when it is empty, means do not enable

Reply:
done